sequencer/src/bin/bridge.rs

```diff
@@ -0,0 +1,343 @@
+use anyhow::{bail, ensure, Context};
+use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+use async_std::{sync::Arc, task::sleep};
+use clap::Parser;
+use contract_bindings::fee_contract::FeeContract;
+use es_version::SequencerVersion;
+use ethers::{
+    middleware::{Middleware, SignerMiddleware},
+    providers::Provider,
+    types::{Address, BlockId, U256},
+};
+use futures::stream::StreamExt;
+use jf_primitives::merkle_tree::{
+    prelude::{MerkleProof, Sha3Node},
+    MerkleTreeScheme,
+};
+use sequencer::{
+    eth_signature_key::EthKeyPair,
+    state::{FeeAccount, FeeAmount, FeeMerkleTree},
+    Header,
+};
+use std::time::Duration;
+use surf_disco::{error::ClientError, Url};
+
+type EspressoClient = surf_disco::Client<ClientError, SequencerVersion>;
+
+type FeeMerkleProof = MerkleProof<FeeAmount, FeeAccount, Sha3Node, { FeeMerkleTree::ARITY }>;
+
+/// Command-line utility for working with the Espresso bridge.
+#[derive(Debug, Parser)]
+enum Command {
+    Deposit(Deposit),
+    Balance(Balance),
+    L1Balance(L1Balance),
```

Comment:
Since most of the arguments are the same I wonder why not put the command specific arguments into a `subcommand` and the rest into a top level struct.

Reply:
I don't think any of the arguments are exactly the same across all three commands unfortunately. `mnemonic` and `account_index` are shared by all of them, but they are required for `deposit` and optional for the balance commands (can just specify a public address instead).

I checked and `cast` seems to work the same way (only top-level options are `-h` and `-v`, everything else is under a sub-command), probably for similar reasons.